src/security/CertGadgets.cc

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 1996-2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "Debug.h"
+#include "sbuf/SBuf.h"
+#include "security/CertGadgets.h"
+
+#if USE_OPENSSL
+#if HAVE_OPENSSL_X509V3_H
+#include <openssl/x509v3.h>
+#endif
+#endif
+
+SBuf
+Security::CertIssuerName(const CertPointer &cert)
+{
+    SBuf out;
+#if USE_OPENSSL
+    auto s = X509_NAME_oneline(X509_get_issuer_name(cert.get()), nullptr, 0);
+    if (!s) {
+        const auto x = ERR_get_error();
+        debugs(83, DBG_PARSE_NOTE(2), "WARNING: issuer: " << Security::ErrorString(x));
```

Comment:
Here and in similar places, please use more meaningful level-1 messages than `WARNING: issuer: error`. For example:

```suggestion
        debugs(83, DBG_PARSE_NOTE(2), "WARNING: cannot get certificate issuer: " << Security::ErrorString(x));
```

Ideally, we should avoid reporting context -- that is the responsibility of other code, but we should report what operation has failed.

Reply:
Done in 611b46a. Along with all similar comments in this function and the matching CertSubjectName() function.